File: OPUS-tauri-app/src/types/workset.ts
export type WorksetProfile = {
  id: string;
  name: string;
  repos: string[];
  workspace_ids: string[];
  defaults?: WorksetDefaults;
  created_at: string;
  updated_at: string;
};

export type WorksetDefaults = {
  base_branch?: string;
  default_remote?: string;
  workspace_root?: string;
};

File: OPUS-tauri-app/src-tauri/src/types/workset.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct WorksetProfile {
    pub id: String,
    pub name: String,
    #[serde(default)]
    pub repos: Vec<String>,
    #[serde(default)]
    pub workspace_ids: Vec<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub defaults: Option<WorksetDefaults>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct WorksetDefaults {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub base_branch: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub default_remote: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub workspace_root: Option<String>,
}
